Sources/Sentry/SentryFileManager.m

```diff
@@ -639,8 +639,7 @@ + (BOOL)createDirectoryAtPath:(NSString *)path withError:(NSError **)error
 
 - (void)createPathsWithOptions:(SentryOptions *_Nonnull)options
 {
-    NSString *cachePath
-        = NSSearchPathForDirectoriesInDomains(NSCachesDirectory, NSUserDomainMask, YES).firstObject;
+    NSString *cachePath = options.cacheDirectoryPath;
```

Comment:
`m`: Please add a test to the SentryFileManagerTests to ensure that the SentyrFileManager uses the path of the options. That's the primary use case of this PR, and I think it's worth adding a test for it.